### ISSUE TITLE
Extending ctc_batch_cost to handle RaggedTensors

### ISSUE THREAD

**Issue by sayakpaul** [tensorflow/tensorflow#41536]
### System information

-   **Have I written custom code (as opposed to using a stock example script
    provided in TensorFlow)**: Yes
-   **OS Platform and Distribution (e.g., Linux Ubuntu 16.04)**: Colab
-   **Mobile device (e.g. iPhone 8, Pixel 2, Samsung Galaxy) if the issue
    happens on a mobile device**:
-   **TensorFlow installed from (source or binary)**:
-   **TensorFlow version (use command below)**:
-   **Python version**:
-   **Bazel version (if compiling from source)**:
-   **GCC/Compiler version (if compiling from source)**:
-   **CUDA/cuDNN version**:
-   **GPU model and memory**:
-   **Exact command to reproduce**:

You can collect some of this information using our environment capture script:

https://github.com/tensorflow/tensorflow/tree/master/tools/tf_env_collect.sh

You can obtain the TensorFlow version with:

```bash
python -c "import tensorflow as tf; print(tf.version.GIT_VERSION, tf.version.VERSION)"
```

### Describe the problem
This is with respect to https://github.com/tensorflow/tensorflow/issues/41276. Basically, I am trying to extend this [official Keras example on OCR](https://colab.research.google.com/github/keras-team/keras-io/blob/master/examples/vision/ipynb/captcha_ocr.ipynb) to the [IAM dataset](http://www.fki.inf.unibe.ch/DBs/iamDB/iLogin/index.php) which consists of images of handwritten characters. Here are some samples from the dataset:

![image](https://user-images.githubusercontent.com/22957388/87866184-23261180-c99c-11ea-8c98-1f9e4b6269c5.png)

 In order for the CTC loss to work (calculated with `ctc_batch_cost`) the ground truth labels and the targets should not be variable-length sequences (`RaggedTensors`s) which is not the case here ([reference](https://github.com/tensorflow/tensorflow/issues/41276#issuecomment-659466273)). So my question is how to extend `ctc_batch_cost` in this case. 

### Source code / logs
[Colab Notebook](https://colab.research.google.com/gist/amahendrakar/71ce949e24f2ac3a4c8ad6ad71590a04/41276-2-3.ipynb)

Cc: @alextp

**Comment by sayakpaul** [comment]
@nikitamaia any updates on this?

**Comment by menonnsl** [comment]
@nikitamaia @sayakpaul

**Comment by sayakpaul** [comment]
@menonnsl here's how I tackled it - 

* Your inputs can have dynamic shapes. The catch is within a batch all the samples should be of the same length. 
* So, while processing each batch I took the highest length of the sample from the current batch and padded the other samples (belonging to the current batch) to that length. You could also compute the global maximum sequence length and pad the other samples to that length. 

If you process your data in this way, you should hopefully be able to start the training.
